### DISCUSSION TITLE
read options from yaml file instead of passed options

### DISCUSSION THREAD

**Discussion by netbe** [erikhuda/thor#822] [Ideas]
Hi,

I did manage to have a subclass of Thor read a config YAML file doing something like this:

```
  private

  def options
    original_options = super
    return original_options unless File.exists?(".babelish")
    defaults = ::YAML.load_file(".babelish") || {}
    Thor::CoreExt::HashWithIndifferentAccess.new(defaults.merge(original_options))
  end
```

this works well until you use required options and then the configuration is not reqd. So i guess my coed should be put elsewhere before required options are checked. Any advice where that may be?

Thanks

**Comment by ralph-tice** [comment]
I think foreman is the original source for this code block: https://github.com/ddollar/foreman/commit/7f8ca7da39df27ea627e3eecb8c65d39d7ed81c7

It would be nice to have first class support for config files.

In actual usage (a class deriving from Thor) I had to do this, which still breaks required options:

```
no_commands {
  def options
        require 'active_support/core_ext/hash/keys'
        original_options = super
        return original_options unless File.exists?(File.expand_path("~/.grout"))
        defaults = ::YAML.load_file(File.expand_path("~/.grout")) || {}
        Thor::CoreExt::HashWithIndifferentAccess.new(defaults.merge(original_options))
      end
    }
```
